score-server/src/main/java/bio/overture/score/server/repository/s3/S3DownloadService.java

```diff
@@ -91,10 +97,14 @@
   private URLGenerator urlGenerator;
   @Autowired
   private PartCalculator partCalculator;
+  @Autowired
+  private MetadataService metadataService;
 
   @Override
   public ObjectSpecification download(String objectId, long offset, long length, boolean forExternalUse) {
     try {
+      checkAnalysisState(metadataService.getEntity(objectId));
```

Comment:
please rename this to `checkPublishedAnalysisState`. Likewise with the uploadService, change the name to `checkUnpublishedAnalysisState`.